PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/controller_class/configuration_controller.cpp

```diff
@@ -0,0 +1,171 @@
+ * 
+ * @param registerNumber -> Taken from the coming TLP
+ * @return int -> 1 for valid Register number, 0 otherwise
+ */
+int ConfigurationController::isValidRegisterNumber(int registerNumber)
+{
+    /* This number will be changed when the PCIE Capability Structure ends */
+    if(registerNumber <= 16)
+        return 1;
+    
+    return 0;
+}
+
+/**
+ * @brief Identifying the Configuration request type
+ * 
+ * @param tlp -> The TLP coming to be handled
+ * @return TLPType -> Configuration Read, or Write
+ */
+TLPType ConfigurationController::getTLPType(TLP * tlp)
+{
+    return tlp->header->TLPtype;
+}
+
+/**
+ * @brief Getting the Data Payload (to be written in a Register) from the coming TLP in case of a Configuration Write request
+ * 
+ * @param tlp -> The TLP coming to be handled
+ * @return unsigned int -> Data inside the TLP
+ */
+unsigned int ConfigurationController::getTLPData(TLP * tlp)
+{
+    boost::dynamic_bitset<> dataPayloadBits;
+
+    unsigned int dataPayloadUint;
+
+    dataPayloadBits = tlp -> dataPayload;
+
+    dataPayloadUint = convertToUnsignedInt(dataPayloadBits);
+
+    return dataPayloadUint;
+}
+
+/**
+ * @brief Converting the coming data inside the TLP from dynamic_bitset to unsigned int
+ * 
+ * @param bits -> Data Payload inside the TLP
+ * @return unsigned int -> the data converted into unsigned int
+ */
+unsigned int ConfigurationController::convertToUnsignedInt(boost::dynamic_bitset<> bits)
+{
+    unsigned int uintValue = static_cast<unsigned int>(bits.to_ulong());
+
+    return uintValue;
+}
+
+/**
+ * @brief Doing the exact opposite of the above
+ * 
+ * @param uintValue 
+ * @return boost::dynamic_bitset<> 
+ */
+boost::dynamic_bitset<> ConfigurationController::convertToBitSet(unsigned int uintValue)
+{
+    boost::dynamic_bitset<> bits(uintValue);
+
+    return bits;
+}
+
+/**
+ * @brief Handling the coming Configuration request through:
+ * 
+ * Getting Register Number -> Checking its validness -> if it's not valid, construct a completion TLP with Unsupported request
+ * if valid, continue to the next steps:
+ * Getting TLP Type (Configuration Read, or Write)
+ * a) Configuration Read:
+ * Get the Data to be read from the given Register number -> convert this unsigned int data to dynamic_bitset -> Determine the TLP construction
+ * algorithm to be made (in this case will be CplD) -> Perform this algorithm and return the TLP result
+ * 
+ * b) Configuration Write:
+ * Get Data to be written to the given register from the given TLP -> Perform the write operation inside the regiser, and if it's not a valid
+ * write operation construct a completion TLP with Unsupported request and return it, otherwise, Determine the TLP construction
+ * algorithm to be made (in this case will be Cpl) -> Perform this algorithm and return the TLP result 
+ * 
+ * @param tlp -> TLP to be handled
+ * @return TLP -> the resultant TLP
+ */
+TLP ConfigurationController::handleConfigurationRequest(TLP * tlp)
```

Comment:
I have a question here 
why you created a handler for configuration requests, and still the logic of handling in the Configuration controller itself